its/src/test/java/com/sonar/it/scanner/msbuild/TestUtils.java

```diff
@@ -232,13 +231,22 @@ public static String replaceLtsVersion(String version) {
     return version;
   }
 
-  public static void runNuGet(Orchestrator orch, Path projectDir, String... arguments) {
+  public static void runNuGet(Orchestrator orch, Path projectDir, Boolean useDefaultVSCodeMSBuild, String... arguments) {
     Path nugetPath = getNuGetPath(orch);
-
-    int r = CommandExecutor.create().execute(Command.create(nugetPath.toString())
+    int r = 1;
+    var nugetRestore = Command.create(nugetPath.toString())
       .addArguments(arguments)
-      .addArguments("-MSBuildPath", TestUtils.getMsBuildPath(orch).getParent().toString())
-      .setDirectory(projectDir.toFile()), 300 * 1000);
+      .setDirectory(projectDir.toFile());
+
+    if (useDefaultVSCodeMSBuild)
+    {
+      r = CommandExecutor.create().execute(nugetRestore, 300 * 1000);
+    }
+    else
+    {
+      r = CommandExecutor.create().execute(nugetRestore
+        .addArguments("-MSBuildPath", TestUtils.getMsBuildPath(orch).getParent().toString()),300 * 1000);
+    }
```

Comment:
It is possible to remove some more code :)
```suggestion
    if(!useDefaultVSCodeMSBuild) {
      nugetRestore = nugetRestore.addArguments("-MSBuildPath", TestUtils.getMsBuildPath(orch).getParent().toString());
    }

    r = CommandExecutor.create().execute(nugetRestore, 300 * 1000);
```